### ISSUE TITLE
server found first/including player

### ISSUE THREAD

**Issue by Kyrasq** [Unprotesting/Auto-Tune#102]
**Describe the bug**

first off hi doing this of  the copy and paste way second is anything depending of most items you can not buy as it say needs to be found by player or server 

**Steps To Reproduce**

fresh copy of the mod no changes at all on a new server
note I did disable the enchantment shop that's all with I bit of payment and time difference for it to refresh 
better and change quicker sort of thing

**Expected behavior**

should just sell no?

**Minecraft Server Version**
1.19 etc.

1.19.2 i hope? paper im using...

**Auto-Tune-Version**
0.13.0 etc.

what ever was the lates on spigot

**Comment by LcyDev** [comment]
Change the settings `PLAYER` `SERVER` to `NONE` in this file `shops.yml`
      collect-first: NONE

**Comment by Kyrasq** [comment]
non of that works @LcyDev i tryed everything i even re wrote the shop with still the same error i went down to 1.14.0 from 1.14.2 and works on the dot with no issue

**Comment by Kyrasq** [comment]
this is the main error for 25% of boots and when people try and buy stuff 
[18:40:44 ERROR]: Error occurred while enabling Auto-Tune v0.14.2 (Is it up to date?)
unprotesting.com.github.mapdb.DBException$WrongConfiguration: Store header checksum, disabled in store, but enabled in configuration
        at unprotesting.com.github.mapdb.StoreDirectAbstract.fileHeaderCheck(StoreDirectAbstract.kt:147) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect.<init>(StoreDirect.kt:114) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect$Companion.make(StoreDirect.kt:57) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect$Companion.make$default(StoreDirect.kt:56) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.DBMaker$Maker.make(DBMaker.kt:450) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.data.Database.createDb(Database.java:201) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.data.Database.<init>(Database.java:59) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.AutoTune.onEnable(AutoTune.java:38) ~[Auto-Tune-0.14.2.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:371) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:544) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_19_R2.CraftServer.enablePlugin(CraftServer.java:578) ~[paper-1.19.3.jar:git-Paper-379]
        at org.bukkit.craftbukkit.v1_19_R2.CraftServer.enablePlugins(CraftServer.java:492) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:635) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:434) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:301) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1101) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.19.3.jar:git-Paper-379]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
[18:40:44 INFO]: [Auto-Tune] Disabling Auto-Tune v0.14.2

**Comment by Kyrasq** [comment]
oh and i forgot about this message about it failing to boot 
[18:44:53 WARN]: Enabled plugin with unregistered PluginClassLoader Auto-Tune v0.14.2
[18:44:53 INFO]: [Auto-Tune] [INFO] Failed to initialize config.
[18:44:53 ERROR]: Error occurred while enabling Auto-Tune v0.14.2 (Is it up to date?)
unprotesting.com.github.mapdb.DBException$FileLocked: File is already opened and is locked: plugins/Auto-Tune/data.db
        at unprotesting.com.github.mapdb.volume.Volume.lockFile(Volume.java:495) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.volume.MappedFileVol.<init>(MappedFileVol.java:88) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.volume.MappedFileVol$MappedFileFactory.factory(MappedFileVol.java:64) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.volume.MappedFileVol$MappedFileFactory.makeVolume(MappedFileVol.java:38) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect$volume$1.invoke(StoreDirect.kt:77) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect$volume$1.invoke(StoreDirect.kt:16) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect.<init>(StoreDirect.kt:76) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect$Companion.make(StoreDirect.kt:57) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.StoreDirect$Companion.make$default(StoreDirect.kt:56) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.mapdb.DBMaker$Maker.make(DBMaker.kt:450) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.data.Database.createDb(Database.java:201) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.data.Database.<init>(Database.java:59) ~[Auto-Tune-0.14.2.jar:?]
        at unprotesting.com.github.AutoTune.onEnable(AutoTune.java:38) ~[Auto-Tune-0.14.2.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:371) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:544) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at com.rylinaux.plugman.util.PluginUtil.enable(PluginUtil.java:69) ~[PlugManX.jar:?]
        at com.rylinaux.plugman.command.EnableCommand.execute(EnableCommand.java:127) ~[PlugManX.jar:?]
        at com.rylinaux.plugman.PlugManCommandHandler.onCommand(PlugManCommandHandler.java:95) ~[PlugManX.jar:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_19_R2.CraftServer.dispatchCommand(CraftServer.java:929) ~[paper-1.19.3.jar:git-Paper-379]
        at org.bukkit.craftbukkit.v1_19_R2.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64) ~[paper-1.19.3.jar:git-Paper-379]
        at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:264) ~[paper-1.19.3.jar:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:313) ~[?:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:297) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.performChatCommand(ServerGamePacketListenerImpl.java:2288) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$20(ServerGamePacketListenerImpl.java:2248) ~[?:?]
        at net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:59) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1341) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1318) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1311) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1389) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[paper-1.19.3.jar:git-Paper-379]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.19.3.jar:git-Paper-379]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.nio.channels.OverlappingFileLockException
        at sun.nio.ch.FileLockTable.checkList(FileLockTable.java:229) ~[?:?]
        at sun.nio.ch.FileLockTable.add(FileLockTable.java:123) ~[?:?]
        at sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1321) ~[?:?]
        at java.nio.channels.FileChannel.tryLock(FileChannel.java:1191) ~[?:?]
        at unprotesting.com.github.mapdb.volume.Volume.lockFile(Volume.java:490) ~[Auto-Tune-0.14.2.jar:?]
        ... 42 more
[18:44:53 INFO]: [Auto-Tune] Disabling Auto-Tune v0.14.2

**Comment by LcyDev** [comment]
Could you please start using https://paste.gg/ or https://pastebin.com/ or atleast use code blocks (Select & Ctrl+E)

Besides that, could you provide your shops.yml and config.yml?

**Comment by LcyDev** [comment]
This just seems like a misconfiguration

**Comment by Kyrasq** [comment]
https://paste.gg/p/anonymous/d0280efdb2dc4668967fc713fe61caa9/revisions is your past bin for you 
[New folder (2).zip](https://github.com/Unprotesting/Auto-Tune/files/10474730/New.folder.2.zip) is your shop ymal note the name will need to be changed for it to work and the config is unchanged by default and or changed

**Comment by LcyDev** [comment]
Your shops.yml is broken.

**Comment by LcyDev** [comment]
https://paste.gg/p/ConaII_/232fc163fcab42048c4cb46bf7f3e030
 
 Here you go, for now use the shops.yml
 
 I'm going to implement a config entry to set the default value of `collect-first`

**Comment by Kyrasq** [comment]
https://paste.gg/p/anonymous/405e539954554f9a86e0315174673545
i still get a one log error with the server website??

**Comment by LcyDev** [comment]
I don't know if the web-server works.
I tried it and it didn't work but that error didn't show.

Is the port open?
Send your `config.yml`

**Comment by Kyrasq** [comment]
https://github.com/Unprotesting/Auto-Tune/issues/102#issuecomment-1399519227the port is the default config launch first try but sadly im still getting the player first error from your codehttps://github.com/Unprotesting/Auto-Tune/issues/102#issuecomment-1399562940

**Comment by LcyDev** [comment]
Player first error?

**Comment by Kyrasq** [comment]
here how to recreate the error down load the lates jar on a clean server no other plugins or something
[archive-2023-01-30T161625+1100.tar.gz](https://github.com/Unprotesting/Auto-Tune/files/10532486/archive-2023-01-30T161625%2B1100.tar.gz)
here is my completed file  of the  configs and everything when your op-ed or not you can not by items that are server first or player first

**Comment by Kyrasq** [comment]
please note the shop would work on 1.14.0 but if an item is not like the hole shop would not load and it would just say this all shop is loaded and running but the menu wouldn't show

**Comment by Kyrasq** [comment]
![2023-02-02_19 54 10](https://user-images.githubusercontent.com/74228066/216317966-b023b6dd-0cf0-4825-aee1-828b3bddbc39.png)
proof of the issue no server logs will show this error also this error is created by default start up config newly created from server

**Comment by LcyDev** [comment]
> https://paste.gg/p/ConaII_/232fc163fcab42048c4cb46bf7f3e030
> 
> Here you go, for now use the `shops.yml`, not the `shops-default.yml`

I'm so sorry this was wrong, you had to use the other one

I updated the file and now it's only one, please try this.

https://paste.gg/p/ConaII_/232fc163fcab42048c4cb46bf7f3e030

**Comment by LcyDev** [comment]
Also, try this build
https://github.com/LcyDev/Auto-Tune/releases/tag/0.14.3

You can add this to your config.yml
```YAML
# If you wish to have collection on your server, enable this.
enable-collection: false
```

**Comment by Kyrasq** [comment]
it is working correctly now but
enable-collection: false
^^^ doggie as broken bone though^^^
having it false doesn't work having true works as normally

**Comment by LcyDev** [comment]
That's weird, wdym by that?
What is not working with that?

**Comment by Schreiner** [comment]
have you manegd to fix it for 1.19.3 to?

**Comment by Kyrasq** [comment]
enable-collection: false <--- stops players to buying stuff
enable-collection: true  <--- lets players buy things 
something is broken in the line of code be default on fresh launch of the plugin hidden is 
enable-collection: false
but writing it in and setting it true as this
enable-collection: true 
allows the server/plugin to run normal setting it back to
enable-collection: false 
will then not let any body buy anything and server player thing isn't working right something..

and yes @Schreiner it works for 1.19.3 but have this in config 

# If you wish to have collection on your server, enable this.
enable-collection: false

^^^mark as true (recommended)^^^

P.S I'm closing this ticket as its fixed BUT would like you to  stream line though the code to see if there is an issue thanks
